restli-server/src/main/java/com/linkedin/restli/internal/server/model/RestLiAnnotationReader.java

```diff
@@ -2595,6 +2595,13 @@ private static void validateFinderMethod(final ResourceMethodDescriptor finderMe
     Method method = finderMethodDescriptor.getMethod();
     Class<?> valueClass = resourceModel.getValueClass();
 
+    if (SingleUnstructuredDataResource.class.isAssignableFrom(resourceModel.getResourceClass())
+        || KeyUnstructuredDataResource.class.isAssignableFrom(resourceModel.getResourceClass()))
+    {
+      throw new ResourceConfigException("Class '" + resourceModel.getResourceClass().getSimpleName()
+          + "' extends the '" + resourceModel.getResourceClass().getSuperclass().getSimpleName() + "' class does not support @Finder methods");
```

Comment:
I just realized that this will not have the intended effect. If class `A` extends some template class, and that extends `KeyUnstructuredDataResource`, then this will print the name of the template class (I think). I would suggest not printing the name of the superclass and instead just append `...because it's an unstructured data resource` at the end of the message.

Reply:
yep. make sense